### ISSUE TITLE
Issues on localhost & Firefox

### ISSUE THREAD

**Issue by WasabiThumb** [gzuidhof/coi-serviceworker#24]
When testing on localhost (WebStorm test server) & Firefox, the service worker never conclusively fails nor succeeds and the site gets stuck in a reboot loop. Also, I noticed that the default ``coepCredentialless`` check will be outdated soon as it is reported working in Firefox >=119. I wasn't sure if this was worthy of a PR so I fixed it in config but you may want to consider adding this in some form.

```js
(() => {

    const DO_NOT_RETRY = "coi-dnr";
    window.coi = {
        shouldRegister: () => {
            if (!window["sessionStorage"]) return false;
            let register = !window.sessionStorage.getItem(DO_NOT_RETRY);
            window.sessionStorage.removeItem(DO_NOT_RETRY);
            return register;
        },
        coepCredentialless: () => {
            if (window["chrome"] !== undefined || window["netscape"] !== undefined) return true;
            const test = /firefox(\/(\d+))?/i;
            try {
                const match = test.exec(window.navigator.userAgent);
                if (match && match[2]) {
                    const n = parseInt(match[2]);
                    if (!isNaN(n) && n >= 119) return true;
                }
            } catch (e) { }
            return false;
        },
        doReload: () => {
            window.sessionStorage.setItem(DO_NOT_RETRY, "true");
            window.location.reload()
        }
    };

})();
```
Also as for immediately quitting if ``sessionStorage`` is not available, [caniuse.net](https://caniuse.com/mdn-api_window_sessionstorage,mdn-http_headers_cross-origin-embedder-policy) reports that for all tracked browsers, there is no case where the COEP header is supported but not ``sessionStorage``.

**Comment by gzuidhof** [comment]
Should be closed with #25 - if not do feel free to re-open.

**Comment by WasabiThumb** [comment]
> Should be closed with #25 - if not do feel free to re-open.

Looks great, thanks! I think that always trying credentialless is a good approach as well over trying to detect browser support ahead of time. With any luck, browsers will eventually catch up on this feature. And in the same vein of wishful thinking, maybe GitHub Pages will start supporting custom headers. 😂
